backend/hitas/services/housing_company.py

```diff
@@ -348,3 +349,16 @@ def find_unregulated_housing_companies_for_reporting() -> list[HousingCompanyWit
         )
         .order_by("-completion_date")
     )
+
+
+def find_housing_companies_for_state_reporting() -> list[HousingCompanyWithStateReportAnnotations]:
+    return list(
+        HousingCompany.objects.annotate(
+            completion_date=max_if_all_not_null(
+                ref="real_estates__buildings__apartments__completion_date",
+                max=datetime.date.max,
+                min=datetime.date.min,
+            ),
+            apartment_count=Count("real_estates__buildings__apartments"),
```

Comment:
This might also be a nice reusable util thingy. It's also used in 4 places already now 😉

Reply:
Added to utils:
```python
max_date_if_all_not_null = partial(max_if_all_not_null, max=datetime.date.max, min=datetime.date.min)
```

Now  you can say:
```python
max_date_if_all_not_null("real_estates__buildings__apartments__completion_date")
```